File: 2419 Longest Subarray With Maximum Bitwise AND/2419.cpp
class Solution {
public:
    int longestSubarray(vector<int>& nums) {
        int ans = 1;
        int maxNum = nums[0];
        int cnt = 0;
        
        for (int num : nums) {
            if (num > maxNum) {
                maxNum = num;
                ans = cnt = 1;
            } else if (num == maxNum) {
                cnt++;
                ans = max(ans, cnt);
            } else {
                cnt = 0;
            }
        }
        
        return ans;
    }
};

File: 2419 Longest Subarray With Maximum Bitwise AND/2419.py
class Solution:
    def longestSubarray(self, nums: List[int]) -> int:
        max_num = max(nums)
        ans = cnt = 0

        for num in nums:
            if num == max_num:
                cnt += 1
                ans = max(ans, cnt)
            else:
                cnt = 0

        return ans
